src/fe/fe_base.C

```diff
@@ -742,7 +742,65 @@ void FEGenericBase<Real>::compute_dual_shape_coeffs ()
 
   //compute dual basis coefficient (dual_coeff)
   dual_coeff.resize(sz, sz);
+  dual_coeff.zero();
+
+  if (fefamily == LAGRANGE && Dim == 1)
+  {
+    // for 1D LAGRANGE, we have pre-computed accurate dual shape coefficients
+    libmesh_error_msg_if(sz > 4, "ERROR:  Unsupported shape function size " << sz);
+    if (sz == 2)
+    {
+      dual_coeff(0,0) = 2.;
+      dual_coeff(0,1) = -1.;
+      dual_coeff(1,0) = -1.;
+      dual_coeff(1,1) = 2.;
+    }
+    if (sz == 3)
+    {
+      dual_coeff(0,0) = 1.5;
+      dual_coeff(0,1) = 0.5;
+      dual_coeff(0,2) = -0.25;
+      dual_coeff(1,0) = 0.5;
+      dual_coeff(1,1) = 1.5;
+      dual_coeff(1,2) = -0.25;
+      dual_coeff(2,0) = -1.;
+      dual_coeff(2,1) = -1.;
+      dual_coeff(2,2) = 1.5;
+    }
+    if (sz == 4)
+    {
+      dual_coeff(0,0) = 2.;
+      dual_coeff(0,1) = -0.5;
+      dual_coeff(0,2) = -0.314814814814815;
+      dual_coeff(0,3) = 0.148148148148148;
+      dual_coeff(1,0) = -0.5;
+      dual_coeff(1,1) = 2.;
+      dual_coeff(1,2) = 0.148148148148148;
+      dual_coeff(1,3) = -0.314814814814815;
+      dual_coeff(2,0) = -0.944444444444444;
+      dual_coeff(2,1) = 0.444444444444443;
+      dual_coeff(2,2) = 1.14403292181070;
+      dual_coeff(2,3) = 0.0226337448559681;
+      dual_coeff(3,0) = 0.444444444444444;
+      dual_coeff(3,1) = -0.944444444444444;
+      dual_coeff(3,2) = 0.0226337448559681;
+      dual_coeff(3,3) = 1.14403292181070;
```

Comment:
I'm guessing that these are all rational numbers. It would be neat to see them expressed as such if possible (but that's ok if you don't have an easy way to do that)

Reply:
Yeah, this is coming from inversion and multiplication of a 4x4 matrix, which should be durable to expressed as rational numbers. I can work a little bit more on this

Reply:
This should also be a `switch` statement or at least `else if` should be used. Maybe the compiler is smart enough to optimize this, but since it gets called a lot may as well try to make it as efficient as possible.

Reply:
BTW we've also slightly improved the way that DenseMatrix  can be constructed, so if you end up going this route you can do, e.g.:
```
dual_coeff = 
  DenseMatrix<Real> (2, 2, { 2, -1,
                                              -1, 2});
```
(the first two args are the number of rows and columns, respectively).

Reply:
> BTW we've also slightly improved the way that DenseMatrix can be constructed, so if you end up going this route you can do, e.g.:
> 
> ```
> dual_coeff = 
>   DenseMatrix<Real> (2, 2, { 2, -1,
>                              -1, 2});
> ```
> 
> (the first two args are the number of rows and columns, respectively).

Oh this is neat. Thanks!